### ISSUE TITLE
[e2e-testing/tutorial] improve helm-chart

### ISSUE THREAD

**Issue by evegufy** [eclipse-tractusx/bpdm#795]
### Description

In order to support [e2e-testing](https://github.com/eclipse-tractusx/e2e-testing/tree/main/charts/umbrella) and tutorial / local setup, I suggest some improvements to the helm-chart.

### Acceptance Criteria

- [x] [BLOCKER FOR INTEGRATION WITH UMBRELLA] solve error when installing with charts that also have a postgresql dependency

I encounter the following error when I install the chart with other that also have a postgresql dependency

```
Error: UPGRADE FAILED: template: umbrella/charts/discoveryfinder/charts/postgresql/templates/primary/svc.yaml:11:14: executing "umbrella/charts/discoveryfinder/charts/postgresql/templates/primary/svc.yaml" at <include "common.labels.standard" (dict "customLabels" .Values.commonLabels "context" $)>: error calling include: template: umbrella/charts/bpdm/charts/postgres/charts/common/templates/_labels.tpl:6:27: executing "common.labels.standard" at <include "common.names.name" .>: error calling include: template: umbrella/charts/bpdm/charts/postgres/charts/common/templates/_names.tpl:6:18: executing "common.names.name" at <.Chart.Name>: nil pointer evaluating interface {}.Name
```
Used config:

```
bpdm:
  bpdm-gate:
    postgres:
      nameOverride: "bpdm-postgresql"
  bpdm-pool:
    postgres:
      nameOverride: "bpdm-postgresql"
  bpdm-bridge-dummy:
    enabled: false
  bpdm-cleaning-service-dummy:
    enabled: false
  postgres:
    nameOverride: "bpdm-postgresql"
    enabled: true
```
May be an issue connected to the different postgres (the other charts are on bitnami chart version 12.x.x) version and be addressed by https://github.com/eclipse-tractusx/bpdm/issues/764

- [ ] provide [images](https://hub.docker.com/search?q=tractusx%2Fbpdm) for arm64 cpu arch, [example](https://hub.docker.com/r/tractusx/bpdm-gate/tags) - in addition to amd64 (image doesn't work on newer Macs, this is relevant especially in the tutorial use case)
- [x] set default imagePullPolicy to `IfNotPresent`
- [ ] Suggestion to change the recommended fullnameOverride(s) in the [values.yaml](https://github.com/eclipse-tractusx/bpdm/blob/bpdm-4.0.1/charts/bpdm/values.yaml) to nameOverride(s) - especially in the case when integrating with charts that have the same dependency, unique labels (name) are important

### Additional Information

evaluation done on chart version 4.0.1

cc: @jzbmw

**Comment by almadigabor** [comment]
Hey @evegufy, good suggestions there, even trying to install the chart directly does not really work for me as most of the pods do not reach the Ready state and restart constantly. This also results in pulling the images on each restart as you described the lack of  default `IfNotPresent` pull policy.

`helm install subchart tractusx-dev/bpdm -n bpdm --create-namespace --version 4.0.1`

```
# kubectl get pods -n bpdm -o wide
NAME                                                    READY   STATUS    RESTARTS        AGE   IP            NODE                   NOMINATED NODE   READINESS GATES
subchart-postgres-0                                     1/1     Running   0               15m   10.42.0.169   lima-rancher-desktop   <none>           <none>
subchart-bpdm-pool-66f67f4499-lgns9                     0/1     Running   1 (3m12s ago)   15m   10.42.0.167   lima-rancher-desktop   <none>           <none>
subchart-bpdm-gate-fb68ff667-8ftgq                      0/1     Running   2 (108s ago)    15m   10.42.0.164   lima-rancher-desktop   <none>           <none>
subchart-bpdm-orchestrator-65675b7977-7kr4f             0/1     Running   2 (93s ago)     15m   10.42.0.165   lima-rancher-desktop   <none>           <none>
subchart-bpdm-bridge-dummy-58b4b9c89-k2758              0/1     Running   2 (78s ago)     15m   10.42.0.163   lima-rancher-desktop   <none>           <none>
subchart-bpdm-cleaning-service-dummy-7964bdbd6f-d27lp   0/1     Running   2 (48s ago)     15m   10.42.0.166   lima-rancher-desktop   <none>           <none>
```

An example on how to implement multiple __docker image architectures__ can be found [here](https://github.com/eclipse-tractusx/policy-hub/blob/main/.github/workflows/policy-hub-service.yml#L77).

**Comment by evegufy** [comment]
> Hey @evegufy, good suggestions there, even trying to install the chart directly does not really work for me as most of the pods do not reach the Ready state and restart constantly. This also results in pulling the images on each restart as you described the lack of default `IfNotPresent` pull policy.
> 
> `helm install subchart tractusx-dev/bpdm -n bpdm --create-namespace --version 4.0.1`
> 
> ```
> # kubectl get pods -n bpdm -o wide
> NAME                                                    READY   STATUS    RESTARTS        AGE   IP            NODE                   NOMINATED NODE   READINESS GATES
> subchart-postgres-0                                     1/1     Running   0               15m   10.42.0.169   lima-rancher-desktop   <none>           <none>
> subchart-bpdm-pool-66f67f4499-lgns9                     0/1     Running   1 (3m12s ago)   15m   10.42.0.167   lima-rancher-desktop   <none>           <none>
> subchart-bpdm-gate-fb68ff667-8ftgq                      0/1     Running   2 (108s ago)    15m   10.42.0.164   lima-rancher-desktop   <none>           <none>
> subchart-bpdm-orchestrator-65675b7977-7kr4f             0/1     Running   2 (93s ago)     15m   10.42.0.165   lima-rancher-desktop   <none>           <none>
> subchart-bpdm-bridge-dummy-58b4b9c89-k2758              0/1     Running   2 (78s ago)     15m   10.42.0.163   lima-rancher-desktop   <none>           <none>
> subchart-bpdm-cleaning-service-dummy-7964bdbd6f-d27lp   0/1     Running   2 (48s ago)     15m   10.42.0.166   lima-rancher-desktop   <none>           <none>
> ```
> 
> An example on how to implement multiple **docker image architectures** can be found [here](https://github.com/eclipse-tractusx/policy-hub/blob/main/.github/workflows/policy-hub-service.yml#L77).

maybe related to the finding from @FaGru3n https://github.com/eclipse-tractusx/bpdm/issues/786

**Comment by nicoprow** [comment]
I provided a fix for this issue here #800 

Seems chart testing did not catch that one. The fixed chart now installs without any other configuration.

Regarding the other suggestions, I think they are good but I will provide such improvements for the next release.

**Comment by nicoprow** [comment]
Nobody working on that now and I don't see the capacity for it in the near future. I close this issue for now and we can reevaluate this workflow feature at a later time
